Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -254,34 +256,37 @@ static OSStatus sampleCallback(CMBufferRef buffer, void* refcon)
         return false;
 
     size_t forwardIndex = 0;
-    // The maximum queue depth possible for out of order frames with either H264 or HEVC is 16, limit looking ahead of 16 frames.
-    for (auto it = m_compressedSampleQueue.begin(); it != m_compressedSampleQueue.end() && forwardIndex < 16; ++forwardIndex, ++it) {
+    for (auto it = m_compressedSampleQueue.begin(); it != m_compressedSampleQueue.end() && forwardIndex < MaximumSlidingWindowLength; ++forwardIndex, ++it) {
         const auto& [sample, flushId] = *it;
         if (flushId != m_flushId)
             return false;
-        if (sample->presentationTime() < time)
+        if (!sample->isNonDisplaying() && sample->presentationTime() < time)
```

Comment:
We are starting to handle non displaying samples, which is fine. But the commit log could be augmented to describe the changes more precisely.